src/main/java/controller/LadderController.java

```diff
@@ -0,0 +1,57 @@
+package controller;
+
+import static java.util.stream.Collectors.*;
+
+import domain.Ladder;
+import domain.LadderHeight;
+import domain.Name;
+import domain.Names;
+import java.util.List;
+import java.util.function.Supplier;
+import utils.NumberGenerator;
+import view.InputView;
+import view.OutputView;
+
+public class LadderController {
+
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final NumberGenerator numberGenerator;
+
+    public LadderController(InputView inputView,
+                            OutputView outputView,
+                            NumberGenerator numberGenerator) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.numberGenerator = numberGenerator;
+    }
+
+    public void run() {
+        Names names = repeat(this::createNames);
+        LadderHeight ladderHeight = repeat(this::createLadderHeight);
+        Ladder ladder = Ladder.create(names.getSize(),ladderHeight,  numberGenerator);
+
+        outputView.printResult(names, ladder);
+    }
+
+    private Names createNames() {
+        List<String> rawNames = inputView.sendNames();
```

Comment:
inputView가 이름을 보낸다..?!? 
```suggestion
        List<String> rawNames = inputView.readNames();
```
가 조금더 자연스러울 것 같아요

Reply:
inputview의 세부 구현을 생각하여 "입력 Stream에 Names를 보낸다"라는 의미에서 어색하지 않지만, 
객체의 역할은 자신의 "어떻게" 할 수 있냐가 아닌, "무엇을"할 수 있냐를 보여준다고 생각해보는 것이 좋을 것 같아요. 

ex) 
자전거를 예를 들어볼게요. 
자전거의 역할은 운동에너지를 받으면 움직이는거에요. 
이경우 인터페이스는

```java
public interface Bike {
  void move(Power power);
} 
```
와 같이 작성될 수 있겠죠. 해당 인터페이스로 작성된 이유는, 객체들의 세상에서, "나는 움직일 수 있다" 라고 나타내기 위함이에요. 이는 자전거의 "무엇을"을 보여주고 있는 셈이죠.

하지만, 자전거가 실제로 움직이는 것은 좀더 복잡합니다. 
"운동에너지를 페달에 전달하면 체인이 돌며, 체인에 맞물려 있는 바퀴가 움직인다."가 어느정도 추상화된 레벨이겠네요. 이부분이 자전거의 "어떻게"인거에요.

해당 로직(sendNames())을 보는 다른 개발자들은, send의 경우 "command"형 메서드라 생각하게됩니다.  
반환값이 없을거라 기대하게되고, 외부 변화를 이끌어 낸다고 생각하죠. 이경우 혼란을 야기할 수 있어서. 피드백 남깁니다 :)

Reply:
기존의 코드는 모든 로직을 알고 있는 제 입장에선 명확하게 느껴지겠지만, 다른 사람들 입장에선 그렇지 않을 수 있겠네요 !!